src/generator/operations.ts

```diff
@@ -423,19 +423,50 @@ function createProtocolRequest(client: string, op: Operation, imports: ImportMan
   const returns = ['*azcore.Request', 'error'];
   let text = `${comment(name, '// ')} creates the ${info.name} request.\n`;
   text += `func (client *${client}) ${name}(${getCreateRequestParametersSig(op)}) (${returns.join(', ')}) {\n`;
+  let includeParse = false;
+  let parseVar = '';
+  // hostURLStr will be used to check for the x-ms-parameterized-host functionality
+  const hostURLStr = <string>op.requests![0].protocol.http!.uri;
+  // for parameterized hosts that only in the format {x} will be ignored and the client url will be used
+  if (!((hostURLStr.match(/{/g) || []).length === 1 && (hostURLStr.match(/}/g) || []).length === 1 && hostURLStr.indexOf('{') === 0 && hostURLStr.indexOf('}') === hostURLStr.length - 1)) {
```

Comment:
Can this be simplified to something like `if (!(hostURLStr[0] === '{' && hostURLStr[hostURLStr.length - 1] === '}'))` or are there other cases.

Reply:
There are other cases like "{accountName}.{dnsSuffix}" Just making it up but it's in the test swaggers

Reply:
Example: https://github.com/Azure/autorest.testserver/blob/master/swagger/custom-baseUrl.json

Reply:
I made a slight change to this condition and updated the comment, PTAL.

Reply:
So I looked at the change, if you think that it's correct to assume that whenever there is only 1 set of `{}` that means we should replace it with the host url, then it looks good to me. My only concern is if there was ever a case where is isn't the host. I did look through the REST API specs, so far it looks like this wont be a problem. Just want to know, how we know that this is correct?

Reply:
The are two buckets of values that can be in `hostURLStr` (which comes from `op.requests![0].protocol.http!.uri`)

- `{whatever}`
- `{whatever}/some/path` (or variants of this with more `{stuff}`)

For the first case (which is what the condition checks for), I don't think it can be anything else but the host URL.

Reply:
Made another change to the condition, PTAL.